### ISSUE TITLE
Warp, Range Suggestion

### ISSUE THREAD

**Issue by ZachZent** [tsunamayo/Skywanderers#1416]
When considering warp, there are two options that in my opinion best conform with the game. First is lanes, stars are connected by preset lanes which ships can travel on. The second is range, ships can warp to any system in a variable range. I personally am on the side of range though both could work well. As such, here is an idea how to implement range in a way that is upgradable, balanceable, and includes an interesting risk factor depending on how deep you want the system.

Note: as stated this is variable. This means that the range can and should be changed as you see fit. Same with recharge rates and everything else. And I know the names are still in flux, so I will be using the term warp.

As stated in the name, your ships warp core, depending on the “level” (or other type of system) will have a set range. Inside that range, you can warp to any star system in that range. Upon exiting, the recharge begins. Now one option is after a certain time, you can warp again. Simple and effective. However there is a more interesting option.

Rather than recharge right away, the recharge range increases over time as you will see in the video. At any time, you can warp into any system within this recharge range with no penalty. You can keep it at that, but we can go even deeper (don’t worry, this is the last inception :D )

You can warp into a system outside of the recharge range, however this isn’t without cost. Depending on the distance outside the recharge range, your ship will take damage. A mix between system, hull, and health. This gives players a more intricate choice. Risk damage to run away from your enemy or get to the other system faster.

There are a lot of pictures here so I'd suggest watching the video for a more complete explanation.
https://www.youtube.com/watch?v=DRyzwitPIX8&feature=youtu.be

**Comment by ultrak2k** [comment]
Would you have a minimum recharge time so you can’t spam warp?

**Comment by ZachZent** [comment]
Oh for sure. All three versions has some type of recharge time. Variable obviously which means different levels for upgrades/changes.

**Comment by Xenoprophet-Nail** [comment]
makes sense to me. since ships would have size constraints it means they'd have a level of classification.. at least as far as jump drives would be concerned anyways

**Comment by tsunamayo** [comment]
Interesting discussion, I had definitely explored some of your idea before, and there is some new interesting one also. For now I will make progress on the gameplay, and see how warp could fit into the game progression. At first I was for a mix of self drive and some gate to be discovered, we will see how it goes! Cheers
